src/ConformalModels/conformal_models.jl

```diff
@@ -0,0 +1,126 @@
+
+Wrapper function to fit the underlying MLJ model and compute nonconformity scores in one single call. This method is only applicable to Transductive Conformal Prediction.
+"""
+function MMI.fit(conf_model::TransductiveConformalModel, verbosity, X, y)
+    fitresult, cache, report = MMI.fit(conf_model.model, verbosity, MMI.reformat(conf_model.model, X, y)...)
+    conf_model.fitresult = fitresult
+    # Use training data to compute nonconformity scores:
+    conf_model.scores = sort(ConformalModels.score(conf_model, X, y), rev=true) # non-conformity scores
+    return (fitresult, cache, report)
+end
+
+"""
+    fit(conf_model::InductiveConformalModel, verbosity, X, y)
+
+Wrapper function to fit the underlying MLJ model. For Inductive Conformal Prediction the underlying model is fitted on the *proper training set*. The `fitresult` is assigned to the model instance. Computation of nonconformity scores requires a separate calibration step involving a *calibration data set* (see [`calibrate!`](@ref)). 
+"""
+function MMI.fit(conf_model::InductiveConformalModel, verbosity, X, y)
+    fitresult, cache, report = MMI.fit(conf_model.model, verbosity, MMI.reformat(conf_model.model, X, y)...)
+    conf_model.fitresult = fitresult
+    return (fitresult, cache, report)
+end
+export fit
+
+# Calibration
+"""
+    calibrate!(conf_model::InductiveConformalModel, Xcal, ycal)
+
+Calibrates a Inductive Conformal Model using calibration data. 
+"""
+function calibrate!(conf_model::InductiveConformalModel, Xcal, ycal)
+    @assert !isnothing(conf_model.fitresult) "Cannot calibrate a model that has not been fitted."
+    conf_model.scores = sort(ConformalModels.score(conf_model, Xcal, ycal), rev=true) # non-conformity scores
+end
+
+export calibrate!
+
+using Statistics
+"""
+    empirical_quantile(conf_model::ConformalModel, coverage::AbstractFloat=0.95)
+
+Computes the empirical quantile `q̂` of the calibrated conformal scores for a user chosen coverage rate `(1-α)`.
+"""
+function empirical_quantile(conf_model::ConformalModel, coverage::AbstractFloat=0.95)
+    @assert 0.0 <= coverage <= 1.0 "Coverage out of [0,1] range."
+    @assert !isnothing(conf_model.scores) "conformal model has not been calibrated."
+    n = length(conf_model.scores)
+    p̂ = ceil(((n+1) * coverage)) / n
+    p̂ = clamp(p̂, 0.0, 1.0)
+    q̂ = Statistics.quantile(conf_model.scores, p̂)
+    return q̂
+end
+export empirical_quantile
+
+# Prediction
+"""
+    MMI.predict(conf_model::ConformalModel, fitresult, Xnew)
+
+Compulsory generic `predict` method of MMI. Simply wraps the underlying model and apply generic method to underlying model.
+"""
+function MMI.predict(conf_model::ConformalModel, fitresult, Xnew)
+    yhat = predict(conf_model.model, fitresult, MMI.reformat(conf_model.model, Xnew)...)
+    return yhat
+end
+
+# Conformal prediction through dispatch:
+"""
+    MMI.predict(conf_model::ConformalModel, Xnew, coverage::AbstractFloat=0.95)
```

Comment:
Type piracy? `MMI.predict(::Model, ...)` will always have a `fitresult` as the second argument here. And your method has an extra parameter `coverage`, which doesn't really fit into the MLJ API at all, unless you make it a hyperparameter (field of the model).

Reply:
Thanks @ablaom. Perhaps one way to avoid this altogether is to treat conformal prediction (set-valued or interval) as a separate functionality, while preserving the model's functionality to produce point predictions. See #14

Reply:
Yep. We have already have `Interval <: Supervised` but maybe can extend that further. Related is helping me out with [this list](https://juliaai.github.io/LearnAPI.jl/dev/operations/#Target-proxies) of "target proxies" (predictions types) for the future.